pandas/io/formats/excel.py

```diff
@@ -721,40 +719,42 @@ def _format_hierarchical_rows(self):
                         fill_value=levels._na_value,
                     )
 
-                    for i in spans:
-                        if spans[i] > 1:
-                            yield ExcelCell(
-                                self.rowcounter + i,
-                                gcolidx,
-                                values[i],
-                                self.header_style,
-                                self.rowcounter + i + spans[i] - 1,
-                                gcolidx,
-                            )
-                        else:
-                            yield ExcelCell(
-                                self.rowcounter + i,
-                                gcolidx,
-                                values[i],
-                                self.header_style,
-                            )
+                    for i, span_val in spans.items():
+                        spans_multiple_cells = span_val > 1
+                        yield ExcelCell(
+                            row=self.rowcounter + i,
+                            col=gcolidx,
+                            val=values[i],
+                            style=self.header_style,
+                            mergestart=(
+                                self.rowcounter + i + spans[i] - 1
```

Comment:
for consistency with L576
```suggestion
                                self.rowcounter + i + spanval - 1
```

this whole block is a duplicate of L567-L578? maybe could refactor to remove duplication in a follow on.